Mathlib/CategoryTheory/Comma.lean

```diff
@@ -63,17 +63,17 @@ variable {T : Type u₃} [Category.{v₃} T]
 /-- The objects of the comma category are triples of an object `left : A`, an object
    `right : B` and a morphism `hom : L.obj left ⟶ R.obj right`.  -/
 structure Comma (L : A ⥤ T) (R : B ⥤ T) : Type max u₁ u₂ v₃ where
-  left : A := by obviously
-  right : B := by obviously
-  Hom : L.obj left ⟶ R.obj right
+  left : A --:= by obviously
+  right : B --:= by obviously
```

Comment:
Can you use `by aesop_cat` here? That is now typically used as replacement for `obviously/tidy`.

Reply:
I know, but I am not sure what `by aesop_cat` would do here because `left/right` are data, not `Prop` fields. Would it have any effect at all?!

Reply:
Surprising that there was `. obviously` there in the first place. I think Scott got a bit carried away. I would suggest just deleting these comments, and leaving a porting note (that the Lean 3 version did not really make sense in the first place). Also in `CommaMorphism` below, except for the equational field `w`.

Reply:
I have just PRed this https://github.com/leanprover-community/mathlib/pull/18440

Reply:
Actually, it turns out that these `by obviously` were necessary in mathlib3. The test mathlib PR just above does not compile. The reason is that for `(co)structured_arrow`, morphisms between initial/terminal objects could be filled automatically `by obviously`. Then, I have added `by aesop_cat` in this mathlib4 PR, and we may hope for the best that when we have limits in mathlib4, `aesop_cat` will be able to fill the obvious fields in `(co)structured_arrow` .

Reply:
I would just add explicit terms wherever needed in mathlib 3; they can't be too complicated right?

Reply:
These terms are easy. Doing so more or less forces the user to use the explicit constructors like `structured_arrow.mk` and `structured_arrow.hom_mk` (rather than providing the relevant fields and relying on automation for the rest). This should eventually make the code cleaner, and easier to maintain/port.